### PR TITLE
Add v1.1 migration guide

### LINKED ISSUE #1260: Add v1.1, initialize migration guide
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

Let's create a migration guide of new & changed documentation, as we've done for past versions: https://docs.getdbt.com/docs/guides/migration-guide/upgrading-to-1-0-0

There shouldn't be any breaking changes for end users. There will be some notable/breaking changes for adapter plugin maintainers and artifact consumers, which we'll want to call out accordingly.

### What part(s) of the page would you like to see updated?

The page needs initialization

### Additional information

Do we like how we've organized these pages in the past? Are there things we want to do differently?

### PR DESCRIPTION
**Pull request by jtcohen6**
resolves https://github.com/dbt-labs/docs.getdbt.com/issues/1260

Let's merge this PR, before all [other v1.1 updates](https://github.com/dbt-labs/docs.getdbt.com/labels/dbt-core%20v1.1.x), since those PRs should update the migration guide.

## Description & motivation

In separate commits, should we want to split any of these off into their own PRs:
- Add v1.1 to version dropdown
- Add "isPrerelease" to VersionContext, and show a warning banner if a user has selected a prerelease version from the dropdown (@JKarlavige would love your eyes on this!)
- Add v1.1 migration guide, with one initial change
- Try using `versionedPages` to show v1.1 migration guide only when v1.1 selected in dropdown (and warn otherwise). This doesn't seem to work as I'd expect, but I'm also not sure it even makes sense for a page like this—we probably do want to show all version guides to all users, no matter which version they're using right now.
- Rename v1.0 migration guide, to follow same `Major.Minor` naming format (with sidebar update + redirect)

## Pre-release docs
Is this change related to an unreleased version of dbt?
- [x] Yes: please
    - update the base branch to `next`
    - add Changelog components: `<Changelog>[New/Changed] in v0.x.0</Changelog>`
    - add links to the "New and changed documentation" section of the latest [Migration Guide](../website/docs/docs/guides/migration-guide)
- [ ] No: please ensure the base branch is `current`
- [ ] Unsure: we'll let you know!

Yes! Let's change the way we do this: https://github.com/dbt-labs/docs.getdbt.com/issues/1261

## Checklist
If you added new pages (delete if not applicable):
- [x] The page has been added to `website/sidebars.js`
- [x] The new page has a unique filename

If you removed existing pages (delete if not applicable):
- [x] The page has been removed from `website/sidebars.js`
- [x] An entry has been added to `_redirects`

### CODE REVIEW

File: website/src/theme/DocPage/index.js
Comment (JKarlavige) [inline website/src/theme/DocPage/index.js]: For this useEffect, it looks like this is causing a `Maximum update depth exceeded` error, due to not having a dependency array.

There's two options to resolve this from what I see. You can either:
- Add the dependency array to the end of this useEffect, making line 75 `}, [dbtVersion])`
- Or, move the if/else statement from this useEffect, into the useEffect below which uses the same `[dbtVersion]` dependency array

This should prevent the useEffect from getting stuck in a constant loop, and only firing when the version has changed.

image

**Review by JKarlavige — CHANGES_REQUESTED** [review]
This is awesome @jtcohen6, nice work! Added one requested change for the useEffect in the DocPage file. Once that is adjusted i'd call this good to go.

File: website/src/theme/DocPage/index.js
Comment (jtcohen6) [reply website/src/theme/DocPage/index.js]: @JKarlavige Ah, so that's what `}, [dbtVersion])` is doing! There's a reason you're the one who gets to do this stuff for real ;)

Thanks for the quick pointer. I like option 2, since that keeps the logic a bit more consolidated. We shouldn't ever have a version that's _both_ a prerelease _and_ within 3 months of EOL.

**Review by JKarlavige — APPROVED** [review]
Verified the depth error is no longer occurring, all looks good to me!

File: website/src/theme/DocPage/index.js
Comment (runleonarun) [inline website/src/theme/DocPage/index.js]: @JKarlavige or @jtcohen6 How can we programmatically add the prerelease version here? 
```suggestion
          isPrereleaseBannerText  : `You are currently viewing v1.1, which is a prerelease of dbt Core. The latest stable version is ${latestStableRelease}`
```

Comment (jtcohen6) [reply website/src/theme/DocPage/index.js]: @runleonarun I think I can do this! comma ça ?

Screenshot 2022-03-29 at 20 38 01

**Comment by jtcohen6** [comment]
Thanks for the feedback both! I'm going to merge this in, in order to
- unblock other v1.1 PRs
- have a "v1.1 landing page" to start linking to from elsewhere

**Comment by jtcohen6** [comment]
Caught one quick thing I needed to fix after merging: https://github.com/dbt-labs/docs.getdbt.com/commit/ebd00034760a0401c52dad44b84f37c27af86b2a

**Comment by JKarlavige** [comment]
Looks great, nice work 👍

**Pull request merged**
